lib/AST/ASTMutations.cpp

```diff
@@ -0,0 +1,72 @@
+#include "mull/AST/ASTMutations.h"
+
+#include <llvm/Support/FileSystem.h>
+
+#include <iostream>
+
+using namespace mull;
+
+ASTMutations::ASTMutations() : storage() {}
+
+int ASTMutations::count() const {
+  int count = 0;
+  for (auto &pair : storage) {
+    /// std::cout << "ASTMutations.count() one ast unit: " << pair.first << "\n";
+    count += storage.at(pair.first).size();
+  }
+  return count;
+};
+
+bool ASTMutations::locationExists(SourceFilePath sourceFile, int line,
+                                  int column) const {
+  assert(llvm::sys::fs::is_regular_file(sourceFile) ||
+         sourceFile == "input.cc");
+
+  int hash = mull::locationHash(line, column);
+
+  if (storage.count(sourceFile) == 0) {
+    return false;
+  }
+
+  const SingleASTUnitMutations &astUnitMutations = storage.at(sourceFile);
+  return (astUnitMutations.count(hash) != 0);
+}
+
+const ASTMutation &ASTMutations::getMutation(const std::string &sourceFile,
+                                             int line, int column) const {
+  assert(llvm::sys::fs::is_regular_file(sourceFile) ||
+         sourceFile == "input.cc");
+
+  assert(storage.count(sourceFile) > 0);
+
+  const SingleASTUnitMutations &astUnitMutations = storage.at(sourceFile);
+
+  MutationLocationHash hash = locationHash(line, column);
+
+  assert(astUnitMutations.count(hash) > 0);
+
+  const ASTMutation &mutation = astUnitMutations.at(hash);
+  return mutation;
+}
+
+void ASTMutations::dump() const {
```

Comment:
I believe that every unit of interest should have a `dump()` method because otherwise we keep repeating the same stuff for debugging all over and over again.

It can also be implemented as `dump(ostream&)` but I am not sure if we need to follow how LLVM does it.